scipp-imaging-2D.ipynb

```diff
@@ -0,0 +1,496 @@
+    "frame_parameters = [(15167, 23563, 64),\n",
+    "                    (24393, 32758, 64),\n",
+    "                    (33365, 40708, 64),\n",
+    "                    (41410, 48019, 64),\n",
+    "                    (49041, 55311, 64),\n",
+    "                    (56077, 59872, 64)]\n",
+    "\n",
+    "# Used to shift the cropped frames so that their bins overlap \n",
+    "# before summing them together into a single frame\n",
+    "frame_shift_increments = [-6630.0, -2420.0, -2253.0, -2095.0, -1946.0, -1810.0]\n",
+    "frame_shifts = [sum(frame_shift_increments[:i + 1]) for i in range(len(frame_shift_increments))]\n",
+    "\n",
+    "# Used to rebin the summed frame in order to\n",
+    "# cut off frames that contain no data\n",
+    "rebin_parameters = (8500, 43000, 64)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import fabio\n",
+    "import numpy as np\n",
+    "import glob\n",
+    "import sys\n",
+    "import csv\n",
+    "\n",
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "from scipp import Dim\n",
+    "\n",
+    "from mantid.simpleapi import *\n",
+    "from mantid.api import AlgorithmManager"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "# Some helper functions\n",
+    "\n",
+    "def read_x_values(tof_file):\n",
+    "    \"\"\"\n",
+    "    Reads the TOF values from the CSV into a list\n",
+    "    \"\"\"\n",
+    "    tof_values = list()\n",
+    "    with open(tof_file) as fh:\n",
+    "        csv_reader = csv.reader(fh, delimiter='\\t')\n",
+    "        next(csv_reader, None)  # skip header\n",
+    "        for row in csv_reader:\n",
+    "            tof_values.append(float(row[1]))\n",
+    "    return tof_values\n",
+    "\n",
+    "\n",
+    "def _load_tiffs(tiff_dir):\n",
+    "    if not os.path.isdir(tiff_dir):\n",
+    "        raise RuntimeError(tiff_dir + \" is not directory\")\n",
+    "    stack = []\n",
+    "    path_length = len(tiff_dir) + 1\n",
+    "    filenames = sorted(glob.glob(tiff_dir + \"/*.tiff\"))\n",
+    "    nfiles = len(filenames)\n",
+    "    count = 0\n",
+    "    print(f\"Loading {nfiles} files from '{tiff_dir}'\")\n",
+    "    for filename in filenames:\n",
+    "        count += 1\n",
+    "        print('\\r{0}: Image {1}, of {2}'.format(filename[path_length:], count, nfiles), end=\"\")\n",
+    "        img = fabio.open(os.path.join(tiff_dir, filename))\n",
+    "        stack.append(np.flipud(img.data))\n",
+    "\n",
+    "    return np.array(stack)\n",
+    "\n",
+    "def tiffs_to_variable(tiff_dir):\n",
+    "    \"\"\"\n",
+    "    Loads all tiff images from the directory into a scipp Variable.\n",
+    "    \"\"\"\n",
+    "    stack = _load_tiffs(tiff_dir)\n",
+    "    return sc.Variable([Dim.Tof, Dim.Spectrum], \n",
```

Comment:
Have you tried `float32`? Maybe more relevant for the benchmark results, but selecting precision is one of the reasons for working with scipp?

Reply:
I have now, and I've added a new table for float32 [here](https://github.com/scipp/scipp/issues/812). It essentially doubles the speed in the slower parts, but the quick ones remain the same or very close